ui/src/Components/ProfilePage.jsx

```diff
@@ -94,13 +94,34 @@ export default class ProfilePage extends React.Component {
         return (
           <Form>
             <div className="right-align" style={{marginRight: '100px'}}>
-            <Link to= {`/profiles/edit/${this.state.profile.result.profile_id}`}className="btn btn-secondary">Edit this profile</Link>
+            <Link to= {`/profiles/edit/${this.state.profile.result.profile_id}`}className="btn btn-secondary">Edit Profile</Link>
             </div>
           </Form>
 
         )
       }
   }
+
+
+  if (this.state.profile && this.state.profile.resultRefs) {
+    const refs = this.state.profile.resultRefs;
+    var refLen = Object.keys(refs).length;
+    var references = [];
+
+    // Builds table with JSON object items passed in
+
+    if (refLen == 0) {
+      references.push(<tr><th>Reference</th><td>Not Provided</td></tr>)
+    }
+
+    for (var i = 0; i < refLen; i++) {
+      references.push(<tr><th>Reference {i+1}</th>
+        <td><div>{refs[i].first_name} {refs[i].last_name}</div>
+        <div>{refs[i].email}</div><div>{refs[i].contact_num.substring(0,3)}-{refs[i].contact_num.substring(3,6)}-{refs[i].contact_num.substring(6,10)} x {refs[i].contact_ext}</div>
```

Comment:
can we call a method somewhere to format this for us? Was a little confused at to what this actually meant.

Reply:
I agree it may be hard to follow, especially because it's outside of the rendered page code.
I'll add some comments instead, since this logic is specific to these reference fields!